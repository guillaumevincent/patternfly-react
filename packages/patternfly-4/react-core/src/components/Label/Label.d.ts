import { HTMLProps, SFC, ReactType, ReactNode } from 'react';
import { OneOf } from '../../typeUtils';

export const LabelVariant: {
  compact: 'compact';
};

export interface LabelProps extends HTMLProps<HTMLLabelElement> {
  children?: ReactNode;
  variant?: OneOf<typeof LabelVariant, keyof typeof LabelVariant>;
}

declare const Label: SFC<LabelProps>;

export default Label;
