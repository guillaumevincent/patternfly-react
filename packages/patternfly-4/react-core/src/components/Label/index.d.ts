export { default as Label, LabelVariant, LabelProps } from './Label';
