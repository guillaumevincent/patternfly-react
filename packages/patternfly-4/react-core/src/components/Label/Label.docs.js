import { Label, LabelVariant } from '@patternfly/react-core';
import SimpleLabel from './examples/SimpleLabel';
import CompactLabel from './examples/CompactLabel';

export default {
  title: 'Label',
  components: {
    Label
  },
  enumValues: {
    'Object.values(LabelVariant)': Object.values(LabelVariant)
  },
  examples: [SimpleLabel, CompactLabel]
};
