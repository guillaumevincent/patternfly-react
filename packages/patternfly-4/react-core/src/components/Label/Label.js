import React from 'react';
import styles from '@patternfly/patternfly-next/components/Label/styles.css';
import PropTypes from 'prop-types';
import { css, getModifier } from '@patternfly/react-styles';

export const LabelVariant = {
  compact: 'compact'
};

const propTypes = {
  /** Content rendered inside the label. */
  children: PropTypes.node,
  /** Additional classes added to the label. */
  className: PropTypes.string,
  /** Adds label variant styles */
  variant: PropTypes.oneOf(Object.values(LabelVariant))
};

const defaultProps = {
  children: '',
  className: '',
  variant: null
};

const Label = ({ className, children, variant, ...props }) => (
  <span {...props} className={css(styles.label, variant && getModifier(styles.modifiers, variant), className)}>
    {children}
  </span>
);

Label.propTypes = propTypes;
Label.defaultProps = defaultProps;

export default Label;
