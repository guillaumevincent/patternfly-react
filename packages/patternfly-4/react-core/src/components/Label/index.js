export { default as Label, LabelVariant } from './Label';
