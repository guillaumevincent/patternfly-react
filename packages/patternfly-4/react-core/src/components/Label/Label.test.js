import React from 'react';
import { shallow } from 'enzyme';
import Label, { LabelVariant } from './Label';

Object.values(LabelVariant).forEach(variant => {
  test(`${variant} Label`, () => {
    const view = shallow(<Label variant={variant}>{variant} Label</Label>);
    expect(view).toMatchSnapshot();
  });
});

test(`simple Label`, () => {
  const view = shallow(<Label>simple Label</Label>);
  expect(view).toMatchSnapshot();
});
