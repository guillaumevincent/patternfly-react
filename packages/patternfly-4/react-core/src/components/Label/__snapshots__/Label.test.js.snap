// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compact Label 1`] = `
.pf-c-label.pf-m-compact {
  display: inline-block;
  padding: 0.25rem 0.5rem 0.25rem 0.5rem;
  font-size: 0.75rem;
  color: #ffffff;
  text-align: center;
  white-space: nowrap;
  background-color: #00659c;
  border-radius: 3px;
}

<span
  className="pf-c-label pf-m-compact"
>
  compact
   Label
</span>
`;

exports[`simple Label 1`] = `
.pf-c-label {
  display: inline-block;
  padding: 0.25rem 0.5rem 0.25rem 0.5rem;
  font-size: 0.875rem;
  color: #ffffff;
  text-align: center;
  white-space: nowrap;
  background-color: #00659c;
  border-radius: 3px;
}

<span
  className="pf-c-label"
>
  simple Label
</span>
`;
